src/ZoneDifficulty.h

```diff
@@ -3,40 +3,160 @@
 
 #include "Player.h"
 #include "Config.h"
+#include "InstanceScript.h"
 #include "ScriptMgr.h"
 #include "ScriptedGossip.h"
 
-struct ZoneDifficultyData
+struct ZoneDifficultyNerfData
 {
     float HealingNerfPct;
     float AbsorbNerfPct;
     float SpellDamageBuffPct;
     float MeleeDamageBuffPct;
-    bool Enabled;
+    int8 Enabled;
+    float HealingNerfPctHard;
+    float AbsorbNerfPctHard;
+    float SpellDamageBuffPctHard;
+    float MeleeDamageBuffPctHard;
 };
 
-const int32 DUEL_INDEX = 0x7FFFFFFF;
-const int32 DUEL_AREA = 2402;       // Forbidding Sea (Wetlands)
+struct ZoneDifficultyHardmodeInstData
+{
+    bool HardmodeOn;
+    bool HardmodePossible;
+};
+
+struct ZoneDifficultyHardmodeMapData
+{
+    uint32 EncounterEntry;
+    uint32 OverrideGO;
+    uint32 RewardType;
+};
+
+struct ZoneDifficultyRewardData
+{
+    uint32 Entry;
+    uint32 Price;
+    uint32 Enchant;
+    uint8 EnchantSlot;
+    uint32 Achievement;
+};
+
+struct ZoneDifficultyHAI
+{
+    uint8 Chance;
+    uint32 Spell;
+    uint8 Target;
+    uint8 TargetArg;
+    std::chrono::milliseconds Delay;
+    std::chrono::milliseconds Cooldown;
```

Comment:
```suggestion
    Milliseconds Delay;
    Milliseconds Cooldown;
```

Reply:
Does this work when passing on timers to the event scheduler?

Reply:
Yea. It is the same thing